tests/test_settings.py

```diff
@@ -19,25 +22,45 @@ def test__parse_pyproject_toml_file_does_not_exist(self) -> None:
 
         assert pyproject_config.toml_config == {}
 
-    def test__parse_pyproject_toml_file_exists(self) -> None:
-        MockedPath = mock.MagicMock(spec=Path)
+    @mock.patch("screenpy.configuration.Path", autospec=True)
+    def test__parse_pyproject_toml_file_exists(self, MockedPath: mock.Mock) -> None:
         MockedPath.cwd.return_value.__truediv__.return_value = MockedPath
         MockedPath.is_file.return_value = True
         test_data = (
             b"[tool.screenpy]\nTIMEOUT = 500"
             b"\n\n[tool.screenpy.stdoutadapter]\nINDENT_SIZE = 500"
         )
         mock_open = mock.mock_open(read_data=test_data)
+        MockedPath.open.side_effect = mock_open.side_effect
+        MockedPath.open.return_value = mock_open.return_value
 
-        with mock.patch("pathlib.Path.open", mock_open):
-            pyproject_config = PyprojectTomlConfig(ScreenPySettings)
-            pyproject_config._parse_pyproject_toml()
+        pyproject_config = PyprojectTomlConfig(ScreenPySettings)
+        pyproject_config._parse_pyproject_toml()
```

Comment:
Are these changes to limit the scope of the `Path` that was mocked?

Reply:
It limits what methods can be called on the mock object.  If you recall a while back the conversation about using `autospec` to narrowly define what methods can be utilized on the mock objects.  This would avoid any potential false positives due to a typo.

```python
# mock has no issues with calling methods that don't actually exist on the original object being mocked
obj = mock.Mock
obj.mispeeled_method()
```

Reply:
The previous code was only and specifically mocking the `Path.open` method with the specially-crafted `mock_open` function from `unittest.mock`. I don't think we would have run into the "mispeeled method" problem here.

Reply:
I'm still not sure it's necessary:

```python
Python 3.12.3 (main, Apr  9 2024, 08:09:14) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from unittest import mock
>>> mo = mock.mock_open(read_data="")
>>> mo.mispeeled_method()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/unittest/mock.py", line 658, in __getattr__
    raise AttributeError("Mock object has no attribute %r" % name)
AttributeError: Mock object has no attribute 'mispeeled_method'
```

I actually think mocking _more_ of `Path` than we need to would make the test less valuable, honestly.

Reply:
If I roll back the mock changes, coverage in `configuration.py` drops to 98%.  
The `test__parse_pyproject_toml_file_does_not_exist` never seems to hit the following line in `PyprojectTomlConfig._parse_pyproject_toml`
```python
        if not pyproject_path.is_file():
            self.toml_config = {}
```

Reply:
I think that missing that line doesn't point to us needing to mock more, i think it means something about our test just above is incorrect. I don't think mocking further is the right answer, we need to look at why the test just above this comment (`test__parse_pyproject_toml_file_does_not_exist`) was not hitting this line of code, when it really ought to be.